src/modules/accounting/prediction.py

```diff
@@ -0,0 +1,94 @@
+from typing import List, Iterable
+from web3.contract import ContractEvent
+from web3.types import Wei, EventData
+from src.typings import BlockStamp
+from variables import REWARDS_HISTORICAL_PERIOD_IN_SECONDS, SECONDS_PER_SLOT
+
+
+class Prediction:
+    def __init__(self, contract_event: ContractEvent):
+        """
+        :param contract_event: ETHDistributed(
+                uint256 indexed reportTimestamp,
+                uint256 preCLBalance,
+                uint256 postCLBalance,
+                uint256 withdrawalsWithdrawn
+                uint256 executionLayerRewardsWithdrawn
+                uint256 postBufferedEther
+            );
+        """
+        self.contract_event = contract_event
+
+    def get_rewards_per_epoch(self, report_timestamp: BlockStamp) -> Wei | Exception:
+        """
+            1. Read previous N report events.
+            2. Get median income
+            3. return median income / epoch_per_day
+
+
+
+            def fetch_predicted_pessimistic_rewards_per_epoch():
+              # Calcs events interval: ref_block - X days duration seconds / 12 ... ref_block
+              # Fetches `ETHDistributed` events from Lido contract
+              # Filters events by timestamp, since there could be missed blocks in slots and we may have captured more blocks
+              pass
+
+            def fetch_predicted_pessimistic_rewards_per_epoch(ref_block):
+              period = fetch_rewards_prediction_period(ref_block) // из контракта
+
+              # Rough estimate, since there could be missing blocks in the slots
+              max_period_block_range = period / SECONDS_PER_SLOT
+
+              from_block = ref_block - max_period_block_range
+              to_block = ref_block
+              events = fetch_eth_distributed_events(from_block, to_block)
+
+              # Narrow the events range
+              events = filter_events_by_timestamp(events)
+
+              return calc_average_rewards_per_epoch_from_events(events)
+        """
+
+        ## TODO: предел CL node на интервал блоков
+        from_block, to_block = self.get_block_for_filtering(report_timestamp['block_number'])
+
+        logs: Iterable[EventData]
+        try:
+            logs = self.contract_event.build_filter().get_logs(
+                from_block=from_block,
+                to_block=to_block,
+            )
+        except Exception as e:
+            return e
```

Comment:
it doesn't look like a "pythonic" way, better just throw it and catch on the top-levels